deploy/azureDeploy.js

```diff
@@ -14,13 +15,18 @@ class AzureDeploy {
 
     Object.assign(
       this,
+      createEventsBindings,
       loginToAzure,
       cleanUpFunctions,
       CreateResourceGroupAndFunctionApp,
       createFunctions
     );
 
     this.hooks = {
+      'before:package:initialize': () => BbPromise.bind(this)
```

Comment:
My initial thought is that it would be great to implement the `sls package` command as part of this PR. That way, we're enabling folks to use `package` and `deploy` as needed, and have things behave as they would expect. Additionally, that seems like it would allow this PR to fully supersede #79.